### PR TITLE
Caffeine

### PR DESCRIPTION
**Pull request by IdaThoft**
A caffeine calculator that displays mg of caffeine in your body when you buy caffeine products.
It also calculates the number of coffee cups corresponding to your caffeine level.
The calculator is only displayed when you buy a caffeine product
Lastly, it also finds the coffee master of the week!

caffeine calculator:
image
image

Coffe master:
image

### CODE REVIEW

**Review by falkecarlsen — CHANGES_REQUESTED** [review]
Nice, more ranking is more better :+1:

File: stregsystem/migrations/0015_product_caffine_content_ml.py [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+        ('stregsystem', '0014_mobilepayment_nullable_customername_20210908_1522'),
+    ]
+
+    operations = [
+        migrations.AddField(
+            model_name='product',
+            name='caffine_content_ml',
+            field=models.FloatField(default=0.0, null=True),
+        ),
+    ]
```
[the remark this answers is not in the document]
Comment (Idadorable) [reply stregsystem/migrations/0015_product_caffine_content_ml.py]: We have create a single file now :)

File: stregsystem/tests.py [hunk 2]
```diff
@@ -1458,3 +1459,78 @@ def test_mobilepaytool_race_error_marx_jdoe(self):
...
+        caffeine_str = "☕☕☕☕☕"
+        caffeine = user.calculate_caffeine_in_body()
+        self.assertEqual(caffeine_str, caffeine_emoji_render(caffeine))
+
+    def test_rich_guy_is_leading_coffee_addict(self):
+        coffee_addict = Member.objects.create(username="Anders", gender='M', balance=100)
+        average_developer = Member.objects.create(username="my-guy", gender='M', balance=50)
+        coffee = Product.objects.create(name="Kaffe☕☕☕", price=1, caffeine_content_mg=71, active=True)
+        # matches coffee id in production. Will be implemented with categories later, when production have a coffee
+        # category
```
[the remark this answers is not in the document]
Comment (Idadorable) [reply stregsystem/tests.py]: We now use pk=6  ✌️

File: stregsystem/views.py [hunk 3]
```diff
@@ -7,9 +8,10 @@
 from django.contrib.auth.decorators import permission_required
 from django.conf import settings
 from django.db.models import Q
-from django import forms
+from django import forms, template
```
Comment (AndersSpringborg) [inline stregsystem/views.py:11]:
```suggestion
from django import forms
```

File: stregsystem/views.py [hunk 4]
```diff
@@ -7,9 +8,10 @@
...
 from django.http import HttpResponsePermanentRedirect, HttpResponseBadRequest
 from django.shortcuts import get_object_or_404, render
+from django.template.defaultfilters import stringfilter
```
Comment (AndersSpringborg) [inline stregsystem/views.py:14]:
```suggestion
```

**Review by AndersSpringborg — CHANGES_REQUESTED** [review]

**Review by AndersSpringborg — APPROVED** [review]
Tror den er der nu! 🤠

**Review by falkecarlsen — APPROVED** [review]
I disliked the addition of the two dependencies, so I refactored the feature and added some more tests :shipit:

**Review by Zaph-x — APPROVED** [review]
☕

**Pull request merged**